### PR TITLE
[collector] Add more fields to k8sattributes metadata

### PR DESCRIPTION
**Pull request by TylerHelmuth**
While doing some experimenting with the presets I noticed that the k8sattributesprocessor was missing some important metadata fields.

### CODE REVIEW

File: charts/opentelemetry-collector/templates/_config.tpl [hunk 1]
```diff
@@ -296,6 +296,10 @@ processors:
         - "k8s.daemonset.name"
         - "k8s.cronjob.name"
         - "k8s.job.name"
+        - "k8s.node.name"
+        - "k8s.pod.name"
+        - "k8s.pod.uid"
```
[the remark this answers is not in the document]
Comment (TylerHelmuth) [reply charts/opentelemetry-collector/templates/_config.tpl:301]: I was doing some work with the preset and wished I could see the node and and pod names of the data I was collecting.  These fields is extremely useful for slicing your data; with them you are able to answer questions like:

- How many traces/metrics/logs am I receiving per node
- How many traces/metrics/logs am I receiving per pod
- Show me when my traces started being generated by my new deployment
- What is the avg span duration per pod
- What is the distribution of spans like per node
- What is the avg value of my custom metric per pod
- How many unique instances of my pod have produced telemetry
- lots more.

Whenever I've wanted to investigate my k8s cluster looking at the traces/metrics/logs by node and by pod have always been my go to.

[the remark this answers is not in the document]
Comment (TylerHelmuth) [reply charts/opentelemetry-collector/templates/_config.tpl:301]: I'm ok adding those as they are default values of the component itself (and I love high cardinality). I am curious how badly this hurts prometheus. In general I'd prefer the chart to favor whats best for helping users troubleshoot issues and understand their systems and not worry about what backends they choose to use.

@open-telemetry/helm-approvers please share your opinions.

[the remark this answers is not in the document]
Comment (puckpuck) [reply charts/opentelemetry-collector/templates/_config.tpl:301]: This is the issue with using a single processor for multiple telemetry types.  High cardinality isn't a concern for tracing backends but is a concern for metric backends.

Specifically about metrics back ends, the number of unique combinations across your attributes/labels for a given metric name is the issue. The pod UID and pod start time, are bound to that pod. Even if you have container restarts within the pod (ie: OOM issue) the UID and start time will remain the same. So adding them when pod name is already a there will not increase the # of timeseries generated and ingested by metric backends like Prometheus.

[the remark this answers is not in the document]
Comment (Allex1) [reply charts/opentelemetry-collector/templates/_config.tpl:301]: I believe `k8s.pod.start_time` is irrelevant if you're using Prometheus scrape to get these metrics and not remotewrite as that's overwritten by scrape time inside Prometheus tsdb.

**Review by Allex1 — APPROVED** [review]

File: charts/opentelemetry-collector/templates/_config.tpl (hunk 1 above)
[the remark this answers is not in the document]
Comment (Allex1) [reply charts/opentelemetry-collector/templates/_config.tpl:301]: Also not sure why both `k8s.pod.name` and `k8s.pod.uid` are needed by default

**Comment by TylerHelmuth** [comment]
@povilasv @Allex1 the argument that @puckpuck made is that uid and start_time have the same cardinality as name. Any time pod name changes so does uid and start_time. If we are good with name we should be good with uid and start_time.

**Comment by TylerHelmuth** [comment]
I agree that adding more attributes to a datapoint will increase its dimensionality. 

Does Prometheus not handle high dimensionality well either?

**Comment by dmitryax** [comment]
As @puckpuck mentioned, both `k8s.pod.uid` and `k8s.pod.start_time` don't change as long as the pod is not recreated, so they are bound to `k8s.pod.name` and don't contribute to the overall cardinality. I'm good with adding them

**Comment by dmitryax** [comment]
> But cardinality is multiplicative....

> If we have pod uid and start time and pod name, then cardinality is:

> unique_labels(pod uid) * unique_labels(pod start time) * unique_labels( pod name)

It's not the case because all three attributes never change individually

**Review by dmitryax — APPROVED** [review]
LGTM

**Review by dmitryax — APPROVED** [review]
LGTM

**Pull request merged**